### PR TITLE
Feature/mds 1441

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
Implemented contact search functionality Nathan did the front end

### CODE REVIEW

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            'items_per_page': pagination_details.page_size,
+            'total': pagination_details.total_results,
+        }
+
+    def apply_filter_and_search(self, args):
+        # Handle ListView request
+        items_per_page = args.get('per_page', 25, type=int)
+        page = args.get('page', 1, type=int)
+        # parse the filter terms
+        first_name_filter_term = args.get('first_name', None, type=str)
```
Comment (Jsyro) [inline python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: Second Parameter is if the key isn't in the dict and returns None by default, so could have empty string has default and the remove all of the first conditions ifs below.

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        last_name_filter_term = args.get('last_name', None, type=str)
+        party_name_filter_term = args.get('party_name', None, type=str)
+        type_filter_term = args.get('type', None, type=str)
+        role_filter_term = args.get('role', None, type=str)
+        email_filter_term = args.get('email', None, type=str)
+        phone_filter_term = args.get('phone_no', None, type=str)
+
+        conditions = []
+        if first_name_filter_term and len(first_name_filter_term.strip()) > 0:
```
Comment (Jsyro) [inline python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: you can add `trim=True` to the add_argument() calls above and it come out stripped.

File: python-backend/app/api/parties/party/resources/party_resource.py [hunk 3]
```diff
@@ -60,7 +61,11 @@ class PartyResource(Resource, UserMixin, ErrorMixin):
     @requires_role_mine_view
     def get(self, party_guid=None):
         if party_guid:
-            party = Party.find_by_party_guid(party_guid)
+            try:
+                party = Party.find_by_party_guid(party_guid)
+            except DBAPIError:
```
Comment (Jsyro) [inline python-backend/app/api/parties/party/resources/party_resource.py]: Is this a common issue, there are alot of find_by_guid method?

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py (hunk 2 above)
Comment (thegentlemanphysicist) [reply python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: that is a useful trick

File: python-backend/app/api/parties/party/resources/party_resource.py (hunk 3 above)
Comment (thegentlemanphysicist) [reply python-backend/app/api/parties/party/resources/party_resource.py]: This was a 500 errror I found through postman that wasn't really part of the ticket.  I put a fix in after chatting with Nathan

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py (hunk 1 above)
Comment (Jsyro) [reply python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: Also just looked, the reqparser also has a default optional parameter if you wanted to do it there as well.

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py (hunk 2 above)
Comment (thegentlemanphysicist) [reply python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: done

File: python-backend/app/api/parties/party/resources/party_advance_search_resource.py (hunk 1 above)
Comment (thegentlemanphysicist) [reply python-backend/app/api/parties/party/resources/party_advance_search_resource.py]: I think it's best to keep the none check in the search function for readability.

File: python-backend/app/api/parties/party/resources/party_resource.py (hunk 3 above)
Comment (thegentlemanphysicist) [reply python-backend/app/api/parties/party/resources/party_resource.py]: The find by guid is mostly used for the add contact drop down.

**Pull request merged**